core/src/test/java/org/jdbi/v3/core/result/TestReducing.java

```diff
@@ -64,8 +67,44 @@ public void testLeftJoinRowView() throws Exception
     }
 
     @Test
-    public void testLeftJoinResultSet() throws Exception
-    {
+    public void testCollectRows() {
+        Iterable<SomethingWithLocations> result = dbRule.getSharedHandle()
+            .createQuery("SELECT something.id, name, location FROM something NATURAL JOIN something_location")
+            .collectRows(Collector.<RowView, Map<Integer, SomethingWithLocations>, Iterable<SomethingWithLocations>>of(
+                    LinkedHashMap::new,
+                    (Map<Integer, SomethingWithLocations> map, RowView rv) ->
+                        map.computeIfAbsent(rv.getColumn("id", Integer.class),
+                                            id -> new SomethingWithLocations(rv.getRow(Something.class)))
+                           .locations
+                           .add(rv.getColumn("location", String.class)),
+                    (a, b) -> {
+                        throw new UnsupportedOperationException("shouldn't use combiner");
```

Comment:
we could make this a public constant somewhere

Reply:
Eh.. feels like overkill for something out of a test.

Reply:
Well, it's not just tests -- most of our collectors have similar mergers.

Reply:
Actually I looked at most of the collectors, and all of them (except this one) do a best-effort merge now.

I think the reasoning was we're providing these collector factory methods for convenience, but there's nothing to stop folks from using them outside of Jdbi. So try to combine containers and play nice.